hpat/tests/test_basic.py

```diff
@@ -264,5 +264,16 @@ def test_impl(n):
         self.assertEqual(count_array_REPs(), 0)
         self.assertEqual(count_parfor_REPs(), 0)
 
+    def test_permuted_array_indexing(self):
+        def test_impl(n):
+            A = np.arange(n)
+            B = np.random.permutation(n)
+            A = A[B]
+            return A.sum()
+
```

Comment:
Needs a test where two arrays are permuted with the same index array, to make sure they are permuted in the same way.